internal/controllers/machinedeployment/machinedeployment_controller.go

```diff
@@ -326,13 +331,17 @@ func (r *Reconciler) getMachineDeploymentsForMachineSet(ctx context.Context, ms
 
 	deployments := make([]*clusterv1.MachineDeployment, 0, len(dList.Items))
 	for idx, d := range dList.Items {
+		d := d
+		log = log.WithValues("MachineDeployment", klog.KObj(&d))
 		selector, err := metav1.LabelSelectorAsSelector(&d.Spec.Selector)
 		if err != nil {
```

Comment:
I'm not sure if for users who read the logs the context of l.338 and l.344 is clear (that we're looking for MDs which match a given MS in the MD reconciler)

Reply:
Makes sense, I will remove them here and in l344.